src/components/namespace_select.tsx

```diff
@@ -8,13 +8,9 @@ import {
   MenuItem,
 } from '@mui/material'
 import Select, { SelectChangeEvent } from '@mui/material/Select'
-import CacheService from 'services/cache_service'
+import React from 'react'
 
-interface NamespaceSelectProps {
-  namespaces: string[]
-  namespacesSelected: string[]
-  setNamespacesSelected(value: string[]): void
-}
+import { NamespacesSelectedContext } from '../pages/SystemAdmin'
```

Comment:
Some stylistic rules we've discussed but haven't yet been written down :-(  Sorry about that...

At any rate, some of this may seem out of scope of this ticket. But since you're touching the files now it makes sense to make these corrections now IMO.

1. Since this is only used by one file, it should be in the directory structure next to that file. So this file belongs in `pages/SystemAdmin`.
2. Since this file's purpose is to export a component, it should be named `NamespaceSelect.tsx`.
3. Just FYI: we're trying to avoid relative imports for anything not in the same directory. So this line, after the move, could be
   ```typescript
   import { NamespacesSelectedContext } from 'pages/SystemAdmin'
   ```
   (because of the export in the `index.ts` file).
4. We no longer do `import React from 'react'` anywhere. Import only the functionality you need, which in this case would be
   ```
   import { useContext } from 'react'
   ```
   and strip off the module name at your `useContext` call site.
5. Lastly, we're avoiding default exports whenever possible. So the last line of this file should be
   ```typescript
   export { NamespaceSelect }
   ```
   with imports in other files corrected as needed.